checkov/common/checks_infra/solvers/connections_solvers/and_connection_solver.py

```diff
@@ -17,24 +18,28 @@ class AndConnectionSolver(ComplexConnectionSolver):
     def __init__(self, solvers: Optional[List[BaseSolver]], operator: str) -> None:
         super().__init__(solvers, operator)
 
-    def get_operation(self, graph_connector: DiGraph) -> Tuple[List[Dict[str, Any]], List[Dict[str, Any]]]:
+    def get_operation(self, graph_connector: DiGraph) -> \
+            Tuple[List[Dict[str, Any]], List[Dict[str, Any]], List[Dict[str, Any]]]:
         if not self.vertices_under_resource_types:
-            return [], []
+            return [], [], []
         subgraph = graph_connector.subgraph(graph_connector)
-        passed, failed = self.run_attribute_solvers(subgraph)
-        failed_ids = [f[CustomAttributes.ID] for f in failed]
-        passed = [p for p in passed if p[CustomAttributes.ID] not in failed_ids]
+        passed, failed, unknown = self.run_attribute_solvers(subgraph)
+        failed_or_unknown_ids = [f[CustomAttributes.ID] for f in itertools.chain(failed, unknown)]
+        passed = [p for p in passed if p[CustomAttributes.ID] not in failed_or_unknown_ids]
 
         for connection_solver in self.get_sorted_connection_solvers():
-            connection_solver.set_vertices(subgraph, failed)
-            passed_solver, failed_solver = connection_solver.get_operation(subgraph)
+            connection_solver.set_vertices(subgraph, failed, unknown)
+            passed_solver, failed_solver, unknown_solver = connection_solver.get_operation(subgraph)
             passed.extend(passed_solver)
             failed.extend(failed_solver)
-            failed_ids.extend([f[CustomAttributes.ID] for f in failed_solver])
-
-        passed = [p for p in passed if p[CustomAttributes.ID] not in failed_ids]
+            unknown.extend(unknown_solver)
+            failed_or_unknown_ids.extend([f[CustomAttributes.ID] for f in itertools.chain(failed_solver, unknown_solver)])
```

Comment:
```suggestion
            failed_or_unknown_ids.extend(f[CustomAttributes.ID] for f in itertools.chain(failed_solver, unknown_solver))
```
`extend()` supports Iterable, so no need to create a list first